cpp/benchmarks/pipelines/registration/Registration.cpp

```diff
@@ -0,0 +1,137 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#include "open3d/pipelines/registration/Registration.h"
+
+#include <benchmark/benchmark.h>
+
+#include <Eigen/Eigen>
+
+#include "open3d/geometry/KDTreeFlann.h"
+#include "open3d/geometry/PointCloud.h"
+#include "open3d/io/PointCloudIO.h"
+#include "open3d/pipelines/registration/TransformationEstimation.h"
+#include "open3d/utility/Console.h"
+
+// Testing parameters:
+// Filename for pointcloud registration data.
+static const std::string source_pointcloud_filename =
+        TEST_DATA_DIR "/ICP/cloud_bin_0.pcd";
+static const std::string target_pointcloud_filename =
+        TEST_DATA_DIR "/ICP/cloud_bin_1.pcd";
+
+static const double voxel_downsampling_factor = 0.05;
+
+// ICP ConvergenceCriteria.
+static const double relative_fitness = 1e-6;
+static const double relative_rmse = 1e-6;
+static const int max_iterations = 1;
+
+// NNS parameter.
+static const double max_correspondence_distance = 0.15;
+
+namespace open3d {
+namespace pipelines {
+namespace registration {
+
+static std::tuple<geometry::PointCloud, geometry::PointCloud> LoadPointCloud(
+        const std::string& source_filename,
+        const std::string& target_filename,
+        const double voxel_downsample_factor) {
+    geometry::PointCloud source;
+    geometry::PointCloud target;
+
+    io::ReadPointCloud(source_filename, source, {"auto", false, false, true});
+    io::ReadPointCloud(target_filename, target, {"auto", false, false, true});
+
+    // Eliminates the case of impractical values (including negative).
+    if (voxel_downsample_factor > 0.001) {
+        source = *source.VoxelDownSample(voxel_downsample_factor);
+        target = *target.VoxelDownSample(voxel_downsample_factor);
+    } else {
+        utility::LogWarning(
+                " VoxelDownsample: Impractical voxel size [< 0.001], skiping "
+                "downsampling.");
+    }
+
+    return std::make_tuple(source, target);
+}
+
+static void BenchmarkRegistrationICPLegacy(
+        benchmark::State& state, const TransformationEstimationType& type) {
+    geometry::PointCloud source;
+    geometry::PointCloud target;
+
+    std::tie(source, target) = LoadPointCloud(source_pointcloud_filename,
+                                              target_pointcloud_filename,
+                                              voxel_downsampling_factor);
+
+    Eigen::Matrix4d init_trans;
+    init_trans << 0.862, 0.011, -0.507, 0.5, -0.139, 0.967, -0.215, 0.7, 0.487,
+            0.255, 0.835, -1.4, 0.0, 0.0, 0.0, 1.0;
+
+    RegistrationResult reg_result(init_trans);
```

Comment:
move the declaration to right before its usage

Reply:
Done.